### PR TITLE
Modularize CR and add volume storage only if set

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

**Review by lrgar — APPROVED** [review]
I have no better ideas for this. I just really hope we get rid of the script soon enough.

File: deploy/install.sh [hunk 1]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
-  classicFullStack:
-    enabled: true
-    tolerations:
-    - effect: NoSchedule
-      key: node-role.kubernetes.io/master
-      operator: Exists
-    env:
-    - name: ONEAGENT_ENABLE_VOLUME_STORAGE
-      value: "${ENABLE_VOLUME_STORAGE}"
+  dynakube=$(
```
Comment (arthfl) [inline deploy/install.sh:109]: Quoting of `$()`

File: deploy/install.sh [hunk 2]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
-      key: node-role.kubernetes.io/master
-      operator: Exists
-    env:
-    - name: ONEAGENT_ENABLE_VOLUME_STORAGE
-      value: "${ENABLE_VOLUME_STORAGE}"
-    args:
-    - --set-host-group="${CLUSTER_NAME}"
+  )
+
+  classicFS=$(
```
Comment (arthfl) [inline deploy/install.sh:122]: Quoting of `$()`

File: deploy/install.sh [hunk 3]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
+    classicFullStack:
+      enabled: true
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+          operator: Exists
+EOF
+  )
+
+  routing=$(
```
Comment (arthfl) [inline deploy/install.sh:134]: Quoting of `$()`

File: deploy/install.sh [hunk 4]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
+    cat <<EOF
+
+    routing:
+      enabled: true
 EOF
+  )
+
+  kubemon=$(
```
Comment (arthfl) [inline deploy/install.sh:142]: Quoting of `$()`

File: deploy/install.sh [hunk 5]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
+    cat <<EOF
+
+    kubernetesMonitoring:
+      enabled: true
+EOF
+  )
+
+  networkZone=$(
```
Comment (arthfl) [inline deploy/install.sh:150]: Quoting of `$()`

File: deploy/install.sh [hunk 6]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
+    cat <<EOF
+
+    networkZone: ${CLUSTER_NAME}
+EOF
+  )
+
+  hostGroup=$(
```
Comment (arthfl) [inline deploy/install.sh:157]: Quoting of `$()`

File: deploy/install.sh [hunk 7]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
+    cat <<EOF
+
+      args:
+        - --set-host-group="${CLUSTER_NAME}"
+EOF
+  )
+
+  agGroup=$(
```
Comment (arthfl) [inline deploy/install.sh:165]: Quoting of `$()`

File: deploy/install.sh [hunk 8]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
+    cat <<EOF
+
+      group: ${CLUSTER_NAME}
+EOF
+  )
+
+  volumeStorage=$(
```
Comment (arthfl) [inline deploy/install.sh:172]: Quoting of `$()`

File: deploy/install.sh [hunk 9]
```diff
@@ -106,60 +106,93 @@ applyDynatraceOperator() {
...
-    enabled: true
-    tolerations:
-    - effect: NoSchedule
-      key: node-role.kubernetes.io/master
-      operator: Exists
-    env:
-    - name: ONEAGENT_ENABLE_VOLUME_STORAGE
-      value: "${ENABLE_VOLUME_STORAGE}"
+  dynakube=$(
+    cat <<EOF
```
Comment (arthfl) [inline deploy/install.sh:110]: Not a big issue, but it's a bit weird that `cat <<EOF` is indented everywhere but the closing `EOF` isn't.
Makes reading the code a bit bumpy.

**Review by arthfl — CHANGES_REQUESTED** [review]

File: deploy/install.sh (hunk 9 above)
Comment (DTMad) [reply deploy/install.sh:110]: You can't intend the closing EOF - otherwise you would loose indentation for the string above

**Pull request merged**